epochStart/genesis/genesisBlockCreator.go

```diff
@@ -322,25 +327,26 @@ func deploySystemSmartContracts(
 	return nil
 }
 
-// setStakingData sets the initial staked values to the staking smart contract
-func setStakingData(
+// setStakedData sets the initial staked values to the staking smart contract
+func setStakedData(
 	txProcessor process.TransactionProcessor,
 	shardCoordinator sharding.Coordinator,
 	initialNodeInfo map[uint32][]*sharding.NodeInfo,
 	stakeValue *big.Int,
 ) error {
 	// create staking smart contract state for genesis - update fixed stake value from all
+	oneEncoded := hex.EncodeToString(big.NewInt(1).Bytes())
 	for i := uint32(0); i < shardCoordinator.NumberOfShards(); i++ {
 		nodeInfoList := initialNodeInfo[i]
 		for _, nodeInfo := range nodeInfoList {
 			tx := &transaction.Transaction{
 				Nonce:     0,
 				Value:     big.NewInt(0).Set(stakeValue),
-				RcvAddr:   vmFactory.StakingSCAddress,
+				RcvAddr:   vmFactory.AuctionSCAddress,
 				SndAddr:   nodeInfo.Address(),
 				GasPrice:  0,
 				GasLimit:  0,
-				Data:      "stake@" + hex.EncodeToString(nodeInfo.PubKey()),
+				Data:      "stake@" + oneEncoded + "@" + hex.EncodeToString(nodeInfo.PubKey()) + "@" + hex.EncodeToString([]byte("genesis")),
```

Comment:
good catch. for genesis we should call "init"

Reply:
Wouldn't it be more consistent that the first string is not encoded always?

Reply:
it is our specification that in case of deploy the first arguments is the encoded code. changing it means to be changed everywhere, and having it encoded it is a more standardized way of handling deployment

Reply:
ok